OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseUnitWebViewFragment.java

```diff
@@ -102,17 +102,20 @@ public void onReceivedError(WebView view, int errorCode,
                     @TargetApi(Build.VERSION_CODES.M)
                     public void onReceivedHttpError(WebView view, WebResourceRequest request,
                                                     WebResourceResponse errorResponse) {
-                        didReceiveError = true;
-                        switch (errorResponse.getStatusCode()) {
-                            case HttpStatus.FORBIDDEN:
-                            case HttpStatus.UNAUTHORIZED:
-                            case HttpStatus.NOT_FOUND:
-                                EdxCookieManager.getSharedInstance(getContext())
-                                        .tryToRefreshSessionCookie();
-                                break;
+                        // If error occurred for web page request
+                        if (request.getUrl().toString().equals(view.getUrl())) {
```

Comment:
Same logic should be applied in the `onReceivedError` callback for similar behaviour on devices lower than Marshmallow version.